### PR TITLE
feat: add `hypot`, `sinc` and `cosc` operations

### PR DESCRIPTION
**Pull request by avik-pal**

### CODE REVIEW

File: src/enzyme_ad/jax/Passes/LowerEnzymeXLAMathPatterns.td [hunk 1]
```diff
@@ -129,3 +129,75 @@ def : Pat<(TGammaOp:$op $x),
...
+            ),
+            (StableHLO_DivOp
+              (StableHLO_MinOp (StableHLO_AbsOp $x), (StableHLO_AbsOp $y)),
+              (StableHLO_MaxOp (StableHLO_AbsOp $x), (StableHLO_AbsOp $y))
+            )
+          )
+        ),
+        ConstDefaultResultAccuracyAttr
+      )
+    )
```
Comment (giordano) [inline src/enzyme_ad/jax/Passes/LowerEnzymeXLAMathPatterns.td:161]: Sorry, I can't quite read tablegen, does this [handle underflow/overflow](https://github.com/EnzymeAD/Reactant.jl/pull/2880#discussion_r3211595497)?  If it does (I see `max`/`min` and divisions here, so it may do), it'd be good to add to the tests the corner cases to show that underflow/overflow works correctly (tests from https://github.com/EnzymeAD/Reactant.jl/pull/2880 would do it)

Comment (avik-pal) [reply src/enzyme_ad/jax/Passes/LowerEnzymeXLAMathPatterns.td:161]: added those tests as well

**Review by giordano — APPROVED** [review]
Looks good based on the tests, can't understand much of tablegen

**Review by wsmoses — APPROVED** [review]

**Comment by wsmoses** [comment]
cc @ptiede
